src/base-xhr.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var objectMirrors = require('./object-mirrors');
+
+var BaseXHR = function(xhrImpl, onreadystatechange, onprogress, onloadend) {
```

Comment:
if i d do it like that i d need another layer in the xhr-shaper implementation as a proxy between BaseXHR and the real XHR.